MathPortal/MathPortal/Source/Custom/Custom Views/SolutionView.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  SolutionView.swift
+//  MathPortal
+//
+//  Created by Petra Čačkov on 03/10/2019.
+//  Copyright © 2019 Petra Čačkov. All rights reserved.
+//
+
+import UIKit
+
+class SolutionView: UIView {
+    
+    var solution: TaskSolution?
+    private var equationViewsHeight: CGFloat = 0
+    private var ownerInfoViewHeight: CGFloat = 60
+    // NOTE: 2 is just a placeholder since cell already has width
+    override var intrinsicContentSize: CGSize {
+        return CGSize(width: 2, height: equationViewsHeight + ownerInfoViewHeight)
+    }
+    
+    func setupView(solution: TaskSolution) {
+        self.solution = solution
+        removeSubviews()
+        addViews()
+        setupFrame()
+        fetchSolutionOwner(solution: solution)
+        self.invalidateIntrinsicContentSize()
+    }
+    
+    private func removeSubviews() {
+        for view in subviews {
```

Comment:
1. Don't ever do this
2. Use subviews.forEach
3. Don't ever do this

Do not ever ever ever assume that the subviews you have are all "your" subviews. Have another private property like `myViews` and remove only those.